tools/cmd.py

```diff
@@ -0,0 +1,510 @@
+# Copyright (c) 2017-2019 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from datetime import datetime
+from enum import IntEnum
+import os
+import re
+import socket
+import struct
+from tools.scp import SCP
+
+
+def _word(a, b, c, d):
+    """
+    Assembles a word out of four bytes.
+
+    aaaaaaaa------------------------   << 24
+    --------bbbbbbbb----------------   << 16
+    ----------------cccccccc--------   <<  8
+    ------------------------dddddddd   <<  0
+    """
+    return (
+        (a & 0xFF) << 24) | ((b & 0xFF) << 16) | ((c & 0xFF) << 8) | (d & 0xFF)
+
+
+# -----------------------------------------------------------------------------
+
+
+class CMD(IntEnum):
```

Comment:
Not a nice name if we plan on using https://docs.python.org/3.8/library/cmd.html

Reply:
Almost all of the names match ones from the ybug code. Which isn't to say that they shouldn't change.

Reply:
`CMD` &rarr; `SCP_CMD`